ot3/gripper_protocols/IDT_xGen_EZ_gripper_EDITED.py

```diff
@@ -0,0 +1,2794 @@
+    'author': 'Opentrons <protocols@opentrons.com>',
+    'source': 'Test Protocols',
+    }
+
+requirements = {
+    "robotType": "OT-3",
+    "apiLevel": "2.13",
+}
+
+
+def right(s, amount):
+    if s == None:
+        return None
+    elif amount == None:
+        return None
+    s = str(s)
+    if amount > len(s):
+        return s
+    elif amount == 0:
+        return ""
+    else:
+        return s[-amount:]
+
+# SCRIPT SETTINGS
+DRYRUN      = 'NO'          # YES or NO, DRYRUN = 'YES' will return tips, skip incubation times, shorten mix, for testing purposes
+
+# !! DO NOT change this!! Changing to NOMODULES = Yes breaks this protocol.
+NOMODULES   = 'NO'          # YES or NO, NOMODULES = 'YES' will not require modules on the deck and will skip module steps, for testing purposes, if DRYRUN = 'YES', then NOMODULES will automatically set itself to 'NO'
+
+TIPREUSE    = 'YES'          # NO, NYI format for reusing tips
+OFFSET      = 'NO'         # YES or NO, Sets whether to use protocol specific z offsets for each tip and labware or no offsets aside from defaults
+MAG         = 'MAGPLATE'
+
+# PROTOCOL SETTINGS
+SAMPLES     = '8x'          # 8x, 16x, or 24x
+FRAGTIME    = 30            # Minutes, Duration of the Fragmentation Step
+PCRCYCLES   = 4             # Amount of Cycles
+
+# PROTOCOL BLOCKS
+STEP_FRERAT         = 1
+STEP_FRERATDECK     = 1
+STEP_LIG            = 1
+STEP_LIGDECK        = 1
+STEP_POSTLIG        = 1
+STEP_PCR            = 1
+STEP_PCRDECK        = 1
+STEP_POSTPCR1       = 1
+STEP_POSTPCR2       = 0
+
+STEPS = {STEP_FRERAT, STEP_LIG, STEP_POSTLIG, STEP_PCR, STEP_POSTPCR1, STEP_POSTPCR2}
+
+MAG_PLATE_SLOT = 1
+EMPTY_SLOT = 8
+USE_GRIPPER = True
+
+
+def run(protocol: protocol_api.ProtocolContext):
+    global TIPREUSE
+    global DRYRUN
+    global p20_tips
+    global p300_tips
+    global MAG
+    global TIPMIX
+
+    if DRYRUN == 'YES':
+        protocol.comment("THIS IS A DRY RUN")
+    else:
+        protocol.comment("THIS IS A REACTION RUN")
+
+    if all(STEPS) == True:
+        if TIPREUSE =='YES':
+            TIPREUSE = 'YES'
+            protocol.comment("TIP REUSING")
+    else:
+        TIPREUSE = 'NO'
+        protocol.comment("NO TIP REUSING")
+
+    # DECK SETUP AND LABWARE
+    if NOMODULES == 'YES':
+        protocol.comment("THIS IS A NO MODULE RUN")
+        # Don't load labware since the plate will be ,oved from TC to mag plate
+        # sample_plate    = protocol.load_labware('nest_96_wellplate_100ul_pcr_full_skirt', '1') #<--- Actually an Eppendorf 96 well, same dimensions
+        if TIPREUSE == 'NO':
+            reservoir           = protocol.load_labware('nest_12_reservoir_15ml','2')
+        else :
+            reservoir           = protocol.load_labware('nest_96_wellplate_2ml_deep','2')
+        reagent_plate       = protocol.load_labware('opentrons_96_aluminumblock_biorad_wellplate_200ul', '3')
+        tiprack_20          = protocol.load_labware('opentrons_96_filtertiprack_20ul',  '4')
```

Comment:
This protocol also loads OT-2 tips.

And note that there is no labware definition for 20 µl OT-3 tips, even though we have a shelf of them here in the office. I don't know what the deal is with that.